src/main/java/com/google/googleidentity/oauth2/authorizationcode/InMemoryCodeStore.java

```diff
@@ -0,0 +1,60 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.authorizationcode;
+
+
+import com.google.googleidentity.oauth2.request.OAuth2Request;
+import com.google.inject.Singleton;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * InMemory AuthorizationCode store, an implementation for
+ * {@link CodeStore}
+ */
+@Singleton
+final class InMemoryCodeStore implements CodeStore {
+
+    private ConcurrentHashMap<String, OAuth2Request> codeStore
+            = new ConcurrentHashMap<>();
+
+    public InMemoryCodeStore(){}
+
+    @Override
+    public Optional<OAuth2Request> consumeCode(String code) {
+        if(codeStore.containsKey(code)) {
+            OAuth2Request request = codeStore.get(code);
+            codeStore.remove(code);
+            return Optional.ofNullable(request);
+        }
+        else{
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public synchronized boolean setCode(String code, OAuth2Request request) {
```

Comment:
Add a comment to note that the synchronized solution here is just for demo as the server is not distributed, and in real, one probably needs storage level synchronizing operation to ensure the uniqueness of auth code.